camel-idea-plugin/src/main/java/org/apache/camel/idea/util/CamelIdeaUtils.java

```diff
@@ -168,7 +168,7 @@ public static boolean isCameSimpleExpressionUsedAsPredicate(PsiElement element)
                         element = element.getParent();
                     }
                     if (element instanceof PsiMethodCallExpression) {
-                        exp = (PsiMethodCallExpression) element;
+                        exp = PsiTreeUtil.getParentOfType(element.getParent(), PsiMethodCallExpression.class);
```

Comment:
this looks a bit wrong, before it was a type cast, you you get the first parent or grand parents that is a method call.